Srclib.Nuget/Graph/GraphRunner.cs

```diff
@@ -14,185 +14,297 @@
-          {
-            // this is a source code file we want to grap
-            runner._sm = context.Compilation.GetSemanticModel(st, false);
-            runner._path = path;
-            var root = await st.GetRootAsync();
-            runner.Visit(root);
-          }
+            foreach(var r in _refs)
+            {
+                try
+                {
+                    var token = r.Item1;
+                    var definition = r.Item2;
+                    var file = r.Item3;
+                    if (_defined.Contains(definition))
+                    {
+                        var reference = Ref.To(definition).At(file, token.Span);
+                        _output.Refs.Add(reference);
+                    }
+                    else
+                    {
+                        _defined.Add(definition);
+                        //at first encountering
+                        var def = Def.For(symbol: definition, type: "external", name: definition.Name).At(file, token.Span);
+                        //_output.Defs.Add(def);
+                        def.External = true;
+                        var reference = Ref.To(definition).At(file, token.Span);
+                        if ((definition.ContainingAssembly != null) && (definition.ContainingAssembly.Identity != null) && (definition.ContainingAssembly.Identity.Name != null))
+                        {
+                            if (dllToProjectUrl.ContainsKey(definition.ContainingAssembly.Identity.Name + ".dll"))
+                            {
+                                string url = dllToProjectUrl[definition.ContainingAssembly.Identity.Name + ".dll"];
+                                if (projectUrlToRepo.ContainsKey(url))
+                                {
+                                    url = projectUrlToRepo[url];
+                                }
+                                //filter out non cloneable urls
+                                //support only github so far???
+                                if (url.IndexOf("github.com") != -1)
+                                {
+                                    reference.DefRepo = url;
+                                    reference.DefUnit = definition.ContainingAssembly.Identity.Name;
+                                    reference.DefUnitType = "NugetPackage";
+                                }
+                            }
+                        }
+                        _output.Refs.Add(reference);
+                    }
+                }
+                catch (Exception e)
+                {
+
+                }
+            }
         }
-      }
 
-      runner._sm = null;
-      runner._path = null;
-      runner.RunTokens();
-      return runner._output;
-    }
+        internal static void HandleNuspec(string path, string dll)
+        {
+            //for some reason xml parsing libraries doesn't seem to be available for dnx50
```

Comment:
It should be dnxcore50, not dnx50. With introduction of dnx, c# applications can be run on different operating systems with different runtimes. That made things much easier, but unfortunately different runtimes have different libraries with different interfaces, and it makes it harder to write universal code that can be compiled everywhere.

It is also somehow explained here http://stackoverflow.com/questions/31654505/dnx-vs-dnx-core

To say it simple, i didn't find an xml parsing library for runtime we use to run toolchain in linux environment. Anyway, the comment was quite meaningless there, i just threw it away.